### PR TITLE
Call callback function after sending mail

### LINKED ISSUE #988: No E-Mail sent when creating user with "node create-user"
## Steps to Reproduce

1. Run `./bin/shell sharelatex`
2. Change into the script directory `cd /var/www/sharelatex/web/modules/server-ce-scripts/scripts`
3. Create user with node command `node create-user --email=email@example.com`

## Expected Behaviour
A User should be created and a mail with an activation URL should be sent to email@example.com

## Observed Behaviour
There is no email sent. When using the web interface to create a user e-mails are working perfectly fine.

## Technical Info
N/A

### LINKED ISSUE #998: Changing docker-compose.yml after tlmgr install scheme-full
I've followed the instructions here 

`https://github.com/overleaf/overleaf/wiki/Quick-Start-Guide`

for installing a full version of TeXLive into overleaf. After having done 

`docker commit sharelatex sharelatex/sharelatex:with-texlive-full`

you supposed to "edit your docker-compose.yml to use that image". Where do I do that? I found several docker-compose-*.yml files in /usr/local/overleaf/lib, and docker-compose.base.yml looks like the most likely place. However, the "image:" line there references a variable ( ${IMAGE} ), so maybe that isn't where you're supposed to change it. 

How do you tell docker to use a different share-latex container when starting overleaf?

### PR DESCRIPTION
**Pull request by chschenk**
## Description
Call callback function after successfully or unsuccessfully
sending the activation mail.

## Related issues / Pull Requests
Fixes #988

## Contributor Agreement

- [x] I confirm I have signed the [Contributor License Agreement](https://github.com/overleaf/overleaf/blob/master/CONTRIBUTING.md#contributor-license-agreement)

### CODE REVIEW

**Review by senk — APPROVED** [review]
LGTM

**Comment by jdleesmiller** [comment]
Thanks for the PR. @chschenk I am not seeing you on the Contributor License Agreement list. Could you please try again?

**Comment by chschenk** [comment]
I have tried again. Did it work this time?

**Comment by jdleesmiller** [comment]
It did, thanks! Now imported to our internal repo. It should land soon.

File: services/web/app/src/Features/User/UserRegistrationHandler.js [hunk 1]
```diff
@@ -131,10 +131,12 @@ const UserRegistrationHandler = {
                 if (error) {
                   logger.warn({ err: error }, 'failed to send activation email')
                 }
+                callback(null, user, setNewPasswordUrl)
+              },
+              success => {
+                callback(null, user, setNewPasswordUrl)
```
Comment (mserranom) [inline services/web/app/src/Features/User/UserRegistrationHandler.js:137]: `EmailHandler.sendEmail()` is created by calling `util.callbackify()` in the original `async` function:

https://github.com/overleaf/overleaf/blob/e0d5cf4b4268d37819811f063754ad1d1f72e061/services/web/app/src/Features/Email/EmailHandler.js#L8-L9

 From `callbackify` [docs](https://nodejs.org/api/util.html#utilcallbackifyoriginal):

> Takes an async function (or a function that returns a Promise) and returns a function following the error-first callback style, i.e. taking an `(err, value) => ... callback` as the last argument

Which effectively means any callback except the last one will be ignored: in this case, the first callback (the one calling `logger.warn()` when sending the email fails) is ignored. A quick example:

```javascript
const util = require('util');

const fnError = () => Promise.reject()

const callbackFunction = util.callbackify(fnError)

callbackFunction(
    (err) => {
        if (err) console.error("1st callback");
    },
    (err) => { 
        if (err) console.error("2nd callback");
    },
);

// Outputs "2nd callback" only
```

Removing the 2nd callback should fix the problem: the error will still be logged and the callback is invoked regardless of whether `sendEmail` fails or not:

```suggestion
```

**Review by mserranom — CHANGES_REQUESTED** [review]
Thanks for contributing! Left an inline comment with a change request.

File: services/web/app/src/Features/User/UserRegistrationHandler.js (hunk 1 above)
Comment (chschenk) [reply services/web/app/src/Features/User/UserRegistrationHandler.js]: Indeed, i removed the 3 lines. Thanks for reviewing.

**Pull request merged**